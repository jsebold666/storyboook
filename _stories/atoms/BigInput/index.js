import React from 'react';
import { text, selectV2 as select, boolean, object } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import readme from './README.md';
import BigInput from '../../../components/atoms/BigInput';

const component = () => (
  <BigInput
    onChange={action('changed')}
    isDisabled={boolean('isDisabled', false)}
    type={text('type', '')}
    value={text('value', 'value')}
  />
);

export default [readme, component];
