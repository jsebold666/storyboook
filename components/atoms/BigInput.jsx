import React, { Component } from 'react';
import styled from 'styled-components';
import shared from '../consts/shared';

const Input = styled.input`
  font-family: Rubik, sans-serif;
  border: none;
  outline: none;
  height: 90px;
  width: 99%;
  margin-left: 0px;
  margin-right: 0px;
  font-size: 36px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 0.69;
  letter-spacing: normal;
  text-align: center;
  color: ${shared['bluey-grey']};
  background-color: ${props => (props.isDisabled ? '#f1f5f7' : 'white')};

  input:focus {
    color: ${shared['slate']};
  }
`;

export default class BigInput extends Component {
  render() {
    return <Input {...this.props} />;
  }
}
