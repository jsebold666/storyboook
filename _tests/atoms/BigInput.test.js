/* globals mount */
import React from 'react';
import BigInput from '../../components/atoms/BigInput';

describe('Expect <BigInput>', () => {
  it('to render', () => {
    const props = {
      value: 'This is text',
      onChange: () => {}
    };

    const wrapper = mount(<BigInput {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('to render properly with empty', () => {
    const props = {
      value: '',
      onChange: () => {}
    };

    const wrapper = mount(<BigInput {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
