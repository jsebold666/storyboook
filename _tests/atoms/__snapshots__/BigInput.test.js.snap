// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <BigInput> to render 1`] = `
<BigInput
  onChange={[Function]}
  value="This is text"
>
  <BigInput__Input
    onChange={[Function]}
    value="This is text"
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "BigInput__Input-xo6509-0",
            "isStatic": false,
            "lastClassName": "iSeKpq",
            "rules": Array [
              "
  font-family: Rubik, sans-serif;
  border: none;
  outline: none;
  height: 90px;
  width: 99%;
  margin-left: 0px;
  margin-right: 0px;
  font-size: 36px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 0.69;
  letter-spacing: normal;
  text-align: center;
  color: ",
              "#acbec7",
              ";
  background-color: ",
              [Function],
              ";

  input:focus {
    color: ",
              "#4a5365",
              ";
  }
",
            ],
          },
          "displayName": "BigInput__Input",
          "render": [Function],
          "styledComponentId": "BigInput__Input-xo6509-0",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onChange={[Function]}
      value="This is text"
    >
      <input
        className="BigInput__Input-xo6509-0 iSeKpq"
        onChange={[Function]}
        value="This is text"
      />
    </StyledComponent>
  </BigInput__Input>
</BigInput>
`;

exports[`Expect <BigInput> to render properly with empty 1`] = `
<BigInput
  onChange={[Function]}
  value=""
>
  <BigInput__Input
    onChange={[Function]}
    value=""
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "BigInput__Input-xo6509-0",
            "isStatic": false,
            "lastClassName": "iSeKpq",
            "rules": Array [
              "
  font-family: Rubik, sans-serif;
  border: none;
  outline: none;
  height: 90px;
  width: 99%;
  margin-left: 0px;
  margin-right: 0px;
  font-size: 36px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 0.69;
  letter-spacing: normal;
  text-align: center;
  color: ",
              "#acbec7",
              ";
  background-color: ",
              [Function],
              ";

  input:focus {
    color: ",
              "#4a5365",
              ";
  }
",
            ],
          },
          "displayName": "BigInput__Input",
          "render": [Function],
          "styledComponentId": "BigInput__Input-xo6509-0",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onChange={[Function]}
      value=""
    >
      <input
        className="BigInput__Input-xo6509-0 iSeKpq"
        onChange={[Function]}
        value=""
      />
    </StyledComponent>
  </BigInput__Input>
</BigInput>
`;
